### ISSUE TITLE
single slider value not set correctly- for below code 

### ISSUE THREAD

**Issue by arunkumar1162** [angular-slider/ng5-slider#56]
Please try this example and see the ticker.
https://stackblitz.com/edit/ng5-slider-simple-slider-example-azi4cj 

step1: click 'days' button,ticker will show value=1 (correct)
step2: click 'limits' button,ticker will show value=3 (correct)
step3: again step1 (fails)

**Comment by piotrdz** [comment]
It's a tricky situation, where you are changing the options and the model value at the same time.

The slider has built-in checks to ensure that the model value is in correct range according to current options. This is always performed according to the options active at the time of the change, which can backfire as in your example.

One workaround for this is to delay applying the value change:
https://stackblitz.com/edit/ng5-slider-simple-slider-example-clq9zs

**Comment by piotrdz** [comment]
Ah, ignore that. It is a working workaround but it works around a bug that shouldn't be there.

I introduced this bug recently with normalisation and range checking code. Changing model values now depends on options object, so the order of applying changes should always be: 1. options object 2. model values.

The thing is, **Angular doesn't guarantee order when applying binding changes**. The proper way to fix this is to implement custom change detection using `ngOnChanges()`. I just fixed this on master and will release in next version.
